### PR TITLE
build: binder: Improve try_from vec

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: build/binder.rs [hunk 1]
```diff
@@ -109,8 +109,20 @@ pub fn generate<W: Write>(modules: Vec<String>, out: &mut W) {
                 if !((buffer[0] == HEADER[0]) && (buffer[1] == HEADER[1])) {
                     return Err(format!("Message should start with \"BR\" ASCII sequence, received: [{0}({:0x}), {1}({:0x})]", buffer[0], buffer[1]));
                 }
+                if buffer.len() < 10 {
```
Comment (joaoantoniocardoso) [inline build/binder.rs:112]: maybe we can give this `10` offset a name defining it as a local const?

Comment (RaulTrombin) [inline build/binder.rs:112]:
```suggestion
                const MIN_MSG_SIZE : usize = 10;
                if buffer.len() < MIN_MSG_SIZE {
```

File: build/binder.rs [hunk 2]
```diff
@@ -109,8 +109,20 @@ pub fn generate<W: Write>(modules: Vec<String>, out: &mut W) {
...
                 let payload_length = u16::from_le_bytes([buffer[2], buffer[3]]);
+
+                if payload_length as usize + 10 != buffer.len() {
+                    return Err(format!(
+                        "Payload length does not match, expected: {}, received: {}",
+                        payload_length,
+                        buffer.len() - 10
+                    ));
+                }
+
```
Comment (RaulTrombin) [inline build/binder.rs:125]:
```suggestion
                if payload_length as usize + MIN_MSG_SIZE != buffer.len() {
                    let received_payload_lenght = buffer.len() - 10;
                    return Err(format!(
                        "Payload length does not match, expected: {payload_length}, received: {received_payload_lenght}"
                    ));
                }

```

**Review by RaulTrombin — COMMENTED** [review]
Just some suggestions

File: build/binder.rs (hunk 1 above)
Comment (patrickelectric) [reply build/binder.rs]: Done.

Comment (patrickelectric) [reply build/binder.rs]: Done.

File: build/binder.rs (hunk 2 above)
Comment (patrickelectric) [reply build/binder.rs]: Moved to use MIN_MSG_SIZE, I see not point of creating an intermediary variable for a print.

**Review by RaulTrombin — APPROVED** [review]

**Review by joaoantoniocardoso — APPROVED** [review]

**Pull request merged**